### PR TITLE
KOA-5828 Remove static colours in compose

### PR DESCRIPTION
**Pull request by marianeum**
Colours are only accessed semantically in compose - we can clean up the static colours to avoid accidental usage. Also fixed two tokens being accidentally treated as dynamic, although they were static.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColors.kt [hunk 1]
```diff
@@ -64,10 +64,6 @@ public class BpkColors private constructor(
   public val surfaceDefault: Color,
   public val surfaceElevated: Color,
   public val surfaceHighlight: Color,
-  @Deprecated(DEPRECATION_MESSAGE)
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColors.kt:67]: these are actually old, static colours that were treated as dynamic due to the detection for what's a dynamic token was wrong

File: buildSrc/src/main/kotlin/net/skyscanner/backpack/tokens/BpkColors.kt [hunk 2]
```diff
@@ -48,7 +48,7 @@ object BpkColor {
 
     override fun invoke(source: Map<String, Any>): BpkColors =
       parseColors(source, resolveReferences = false) {
-        it.name == it.defaultReference && !it.isMarcomms()
+        !it.isSemanticColor() && !it.isMarcomms()
```
Comment (marianeum) [inline buildSrc/src/main/kotlin/net/skyscanner/backpack/tokens/BpkColors.kt:51]: this is where the bug was

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColors.kt (hunk 1 above)
Comment (bvitaliyg) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColors.kt:67]: I think we need to clean up the foundations and remove all these custom tricks eventually

File: buildSrc/src/main/kotlin/net/skyscanner/backpack/tokens/BpkColors.kt [hunk 3]
```diff
@@ -178,29 +173,6 @@ private fun deprecationProperty(): PropertySpec =
 @OptIn(ExperimentalStdlibApi::class)
 private fun String.toHexColor() = uppercase().run { "0x${substring(7)}${substring(1, 7)}" }
 
-private fun toStaticCompose(
```
Comment (bvitaliyg) [inline buildSrc/src/main/kotlin/net/skyscanner/backpack/tokens/BpkColors.kt:181]: So, we're not generating static compose colors at all, right?

**Review by bvitaliyg — APPROVED** [review]
Some quesntions

File: buildSrc/src/main/kotlin/net/skyscanner/backpack/tokens/BpkColors.kt (hunk 3 above)
Comment (marianeum) [reply buildSrc/src/main/kotlin/net/skyscanner/backpack/tokens/BpkColors.kt:181]: nope - we're meant to only use semantic ones anyways. they're also not used in the app at all. with the view system we've got historic colour usage, but in compose we've started from scratch

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColors.kt (hunk 1 above)
Comment (marianeum) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColors.kt:67]: eventually. but right now we still have usages of these (in the view system)

**Pull request merged**